Fix `fakeplayer_inv_editor.sc` for 1.21+

SethHu20 opened the pull request:
As of 1.21.5 and Carpet 1.4.169, `fakeplayer_inv_editor.sc` will throw `NullPointerException` when calling `inventory_set()` in various places within the script. It seems that the NBT tag argument that got passed into `inventory_set()` causes this issue. Upon further investigation, my current conclusion is:

1. the NBT tag argument must also include the id of the item
2. custom data should now be placed in the `components.custom_data` tag, otherwise it would be lost

To address the two observations:
1. Barrier and structure void items for this script now have their own version of NBT tags
2. The custom tag is now added and checked in `components.custom_data`. It is possible that this change breaks compatibility to older version, most likely before 1.21 snapshot 24w19a where this tag is added (https://minecraft.wiki/w/Java_Edition_24w19a)

SethHu20 committed:
Fix NBT tags for `global_nope`

Fix `global_nope` NBT data to utilize `components.custom_data` properties as of 1.21.5, inventory_set now uses a more complete NBT tag to successfully add the custom NBT data

Xiang990293 commented:
I'm not sure but I found if I use the script  you provide, though the NullPointerException is not popping up, but will shift the slot in the large chest ui,
For example, the top four-in-a-row slots, seems equipment slot, but if put something in second slot from let, the item will shows on offhand instead of chest.

replaceitem commented:
Makes it work again. The equipment slots @Xiang990293 mentioned I have fixed in #433 along with the datapack folder name changes.

Ghoulboy78 committed:
Updating minimum required carpet version to reflect changes

That way someone using an older version will know why this updated version of the script doesn't work for them
